hystrix-core/src/main/java/com/netflix/hystrix/AbstractCommand.java

```diff
@@ -602,15 +602,23 @@ public void call() {
         final Func1<Throwable, Observable<R>> handleFallback = new Func1<Throwable, Observable<R>>() {
             @Override
             public Observable<R> call(Throwable t) {
-                circuitBreaker.markNonSuccess();
                 Exception e = getExceptionFromThrowable(t);
                 executionResult = executionResult.setExecutionException(e);
+
                 if (e instanceof RejectedExecutionException) {
+                    circuitBreaker.markNonSuccess();
                     return handleThreadPoolRejectionViaFallback(e);
                 } else if (t instanceof HystrixTimeoutException) {
+                    circuitBreaker.markNonSuccess();
                     return handleTimeoutViaFallback();
                 } else if (t instanceof HystrixBadRequestException) {
-                    return handleBadRequestByEmittingError(e);
+                    HystrixBadRequestException hbre = (HystrixBadRequestException) e;
```

Comment:
Should be `t` instead of `e` since the else block does an instance of check on `t`.